src/TreeView/TreeView.tsx

```diff
@@ -22,63 +33,275 @@ export type TreeViewProps = {
+    throw new Error('Element is not a treeitem')
+  }
+
+  switch (element.getAttribute('aria-expanded')) {
+    case 'true':
+      return 'open'
+    case 'false':
+      return 'closed'
+    default:
+      return 'end'
+  }
+}
+
+function getVisibleElement(element: HTMLElement, direction: 'next' | 'previous'): HTMLElement | undefined {
+  const root = element.closest('[role=tree]')
+
+  if (!root) return
+
+  const walker = document.createTreeWalker(root, NodeFilter.SHOW_ELEMENT, node => {
+    if (!(node instanceof HTMLElement)) return NodeFilter.FILTER_SKIP
+    return node.getAttribute('role') === 'treeitem' ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_SKIP
+  })
+
+  let current = walker.firstChild()
+
+  while (current !== element) {
+    current = walker.nextNode()
+  }
+
+  let next = direction === 'next' ? walker.nextNode() : walker.previousNode()
+
+  // If next element is nested inside a collapsed subtree, continue iterating
+  while (next instanceof HTMLElement && next.parentElement?.closest('[role=treeitem][aria-expanded=false]')) {
+    next = direction === 'next' ? walker.nextNode() : walker.previousNode()
+  }
+
+  return next instanceof HTMLElement ? next : undefined
+}
+
+function getFirstChildElement(element: HTMLElement): HTMLElement | undefined {
+  const firstChild = element.querySelector('[role=treeitem]')
+  return firstChild instanceof HTMLElement ? firstChild : undefined
+}
+
+function getParentElement(element: HTMLElement): HTMLElement | undefined {
+  const group = element.closest('[role=group]')
+  const parent = group?.closest('[role=treeitem]')
+  return parent instanceof HTMLElement ? parent : undefined
+}
+
+function getFirstElement(element: HTMLElement): HTMLElement | undefined {
+  const root = element.closest('[role=tree]')
+  const first = root?.querySelector('[role=treeitem]')
+  return first instanceof HTMLElement ? first : undefined
+}
+
+function getLastElement(element: HTMLElement): HTMLElement | undefined {
+  const root = element.closest('[role=tree]')
+  const items = Array.from(root?.querySelectorAll('[role=treeitem]') || [])
+
+  // If there are no items, return undefined
+  if (items.length === 0) return
+
+  let index = items.length - 1
+  let last = items[index]
+
+  // If last element is nested inside a collapsed subtree, continue iterating
+  while (
+    index > 0 &&
+    last instanceof HTMLElement &&
+    last.parentElement?.closest('[role=treeitem][aria-expanded=false]')
+  ) {
+    index -= 1
+    last = items[index]
+  }
+
+  return last instanceof HTMLElement ? last : undefined
+}
+
 // ----------------------------------------------------------------------------
 // TreeView.Item
 
 export type TreeViewItemProps = {
   children: React.ReactNode
-  onSelect?: (event: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>) => void
+  defaultExpanded?: boolean
+  onSelect?: (event: React.MouseEvent<HTMLElement> | KeyboardEvent) => void
```

Comment:
When we create stories and/or examples, let's not forget to include examples that show consumers that they have to move focus to the first focusable element in the main content area when they activate a link node.

(Nothing needs to change in this PR)

Reply:
Good callout 👍